devchat/config.py

```diff
@@ -44,14 +44,24 @@ class OpenAIModelConfig(ModelConfig, OpenAIChatParameters):
 class AnthropicModelConfig(ModelConfig, AnthropicChatParameters):
     pass
 
+class GeneralModelConfig(BaseModel, extra='allow'):
+    config: Optional[Any]
+    def __init__(self, **attri):
+        object.__setattr__(self, "config", attri)
+        if "max_input_tokens" not in self.config:
+            self.config["max_input_tokens"] = sys.maxsize
+
+    def dict(self, **_):
+        return self.config
+
+    def __setattr__(self, name, value):
+        self.config[name] = value
+
+    def __getattr__(self, name):
+        if name not in self.config:
+            return None
+        return self.config[name]
 
-class GeneralModelConfig(ModelConfig):
-    max_tokens: Optional[int]
-    stop_sequences: Optional[List[str]]
-    temperature: Optional[float]
-    top_p: Optional[float]
-    top_k: Optional[int]
-    stream: Optional[bool]
```

Comment:
Does Xinghuo support the above parameters? Or, we can just add the minimal necessary parameters of Xinghuo but still keep the extra='forbid'. It is good to keep some bottom-line check.